pkg/configuration/configuration.go

```diff
@@ -280,3 +290,74 @@ func (c *Config) KeyringPrefix() (string, error) {
 	}
 	return h, nil
 }
+
+func (c *Config) CheckForUpdate(out io.Writer, current string) error {
+	// Check if version check is disabled in configuration
+	if c.Meta.DisableVersionCheck {
+		return errors.New("version check disabled")
+	}
+
+	// Check if version check has already been done today
+	yesterday := time.Now().AddDate(0, 0, -1)
+	lastCheck, _ := time.Parse(time.RFC3339Nano, c.Meta.LastChecked)
+	if !lastCheck.Before(yesterday) {
+		return errors.New("version check already done today")
+	}
+
+	// Perform version check
+	const cliReleasesURL = "https://api.github.com/repos/appgate/sdpctl/releases"
+	req, err := http.NewRequest(http.MethodGet, cliReleasesURL, nil)
+	if err != nil {
+		return err
+	}
+	req.Header.Add("Accept", "application/vnd.github+json")
+	client := &http.Client{}
+	res, err := client.Do(req)
```

Comment:
we need to set a timeout here, it's very likely that github.com can be blocked/unreachable for the customers

Reply:
We should probably write a new last check time too, even though the request fails, so we don't retry on every consecutive command when Github is unreachable.